src/sst/elements/vanadis/inst/vzicntr.h

```diff
@@ -0,0 +1,17 @@
+
+#ifndef _H_VANADIS_ZICNTR
+#define _H_VANADIS_ZICNTR
+
+#include <utility>
+
+namespace SST::Vanadis
+{
+    // Tags to be used to with the regfile and VanadisReadCounterInstruction
+    namespace Zicntr {
+        inline constexpr std::integral_constant<uint32_t, 0> CYCLE;
+        inline constexpr std::integral_constant<uint32_t, 1> TIME;
+        inline constexpr std::integral_constant<uint32_t, 2> INSTRET;
+    }
```

Comment:
This really just looks like a class enum type implemented with integral_constant., though I may be missing something deeper.  Is there a reason not to simply use a class enum?

Reply:
They encode the index as a compile time value wherever it's used. So I use them as a tag type to the `Zicntr::VanadisReadCounterInstruction` class template constructor and I don't have to set any explicit class template parameters. 

```c++
template <uint32_t ID> 
struct Foo {
    Foo(const std::integral_constant<uint32_t, ID>, stuff...);
};

std::integral_constant<uin32_t, 7> TAG;

Foo foo(TAG, stuff...); // ID is inferred as 7
```

as opposed to

```c++
enum class E {
    A, B, ...
};

template <E ID>
struct Foo {
    Foo(stuff...);
};

Foo<E::A> foo(stuff...); // have to be explicit here
```

They're used in the code like:

```c++
bundle->addInstruction( new VanadisReadCounterInstruction( INSTRET, ins_address, hw_thr, options, rd ) )
```

If `vanadis` would prefer an enum and to use `VanadisReadCounterInstruction<Enum::INSTRET>(...)` that's fine with me.

Reply:
(they're also serve fine as case statement labels)